src/lib/core/backtrace.c

```diff
@@ -246,11 +246,15 @@ backtrace_frame_resolve(const struct backtrace_frame *frame,
 #ifndef __APPLE__
 	unw_accessors_t *acc = unw_get_accessors(unw_local_addr_space);
 	assert(acc->get_proc_name != NULL);
-	char *proc_name_buf = tt_static_buf();
+	/*
+	 * In case of long name we get truncated version with no demangling.
+	 * Not too bad.
+	 */
+	char proc_name_buf[128];
 	int rc = acc->get_proc_name(unw_local_addr_space, (unw_word_t)frame->ip,
-				    proc_name_buf, TT_STATIC_BUF_LEN, offset,
-				    NULL);
-	if (rc != 0) {
+				    proc_name_buf, sizeof(proc_name_buf),
+				    offset, NULL);
+	if (rc != 0 && rc != UNW_ENOMEM) {
```

Comment:
Checked and added a comment as discussed f2f. Name is truncated in case of UNW_ENOMEM.

It is also documented https://www.nongnu.org/libunwind/man/unw_create_addr_space(3).html